### PR TITLE
Fix batchedsend restart

### PR DESCRIPTION
**Pull request by fjetter**
Supersedes https://github.com/dask/distributed/pull/6266

### CODE REVIEW

**Comment by fjetter** [comment]
Locally it looks like the failing cases with `test_flight_to_executing_via_cancelled_resumed` are caused by https://github.com/dask/distributed/pull/5910

**Review by gjoseph92 — COMMENTED** [review]
I think this PR will be an improvement, as a hotfix for a current deadlock https://github.com/dask/distributed/issues/6228.

However, it introduces a new potential bug. (And BatchedSend already has many existing bugs and foot-guns, enumerated in my review on https://github.com/dask/distributed/pull/5457.)

Are we okay treating this as a hotfix and leaving this multiple-start bug for future work, or do we want to fix it here?

File: distributed/batched.py [hunk 1]
```diff
@@ -125,50 +139,45 @@ def _background_send(self):
...
         If set, raises `tornado.util.TimeoutError` after a timeout.
         """
-        if self.comm is None:
-            return
         self.please_stop = True
         self.waker.set()
-        yield self.stopped.wait(timeout=timeout)
-        if not self.comm.closed():
+        if self._background_task:
+            await self._background_task
```
[the remark this answers is not in the document]
Comment (fjetter) [reply distributed/batched.py:158]: background task doesn't raise any exceptions

**Review by gjoseph92 — COMMENTED** [review]
A couple race conditions I'd like to fix. I find `WorkerStatusPlugin` weird, but can't decide if it's problematic enough to justify delaying this PR if everything else is otherwise fixed.

Also, can you remove this now-irrelevant block: https://github.com/dask/distributed/blob/22868968b39a9b41761595cd61316dadba5c4ec5/distributed/worker.py#L3429-L3431

**Pull request closed without merging**